src/chart/generateCategoricalChart.tsx

```diff
@@ -2151,6 +2151,12 @@ export const generateCategoricalChart = ({
             return [graphicalItem, this.renderActiveBar({ item, childIndex: activeTooltipIndex })];
           }
 
+          if (activeShape) {
+            const activeIndex =
+              element.props.activeIndex !== undefined ? element.props.activeIndex : activeTooltipIndex;
+            return [cloneElement(element, { ...item.props, ...itemEvents, activeIndex }), null, null];
```

Comment:
Just wondering, do you see a reason why activeBar should be treated separately? Could we not treat all activeShapes the same? 

Or is there some specific behaviour the I am not aware of?

Reply:
Addressed here: https://github.com/recharts/recharts/pull/3803/commits/80bd132bfecaee13a84d09b661a86a59c425a0eb